### PR TITLE
qt6.qtsvg: add mingw support

### PR DESCRIPTION
**Pull request by seanybaggins**
Part of a larger effort to add mingw support to qtmultimedia
Relevant for https://github.com/NixOS/nixpkgs/issues/274274
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingw32.qt6.qtsvg

**Comment by seanybaggins** [comment]
@wegank @Artturin @bjornfor

**Comment by seanybaggins** [comment]
Going to change ordering of packages so rebuild count isn't so high. Standby.

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingw32.qt6.qtsvg

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingwW64.qt6.qtsvg

**Comment by bjornfor** [comment]
Nit: please end full sentences in commit message bodies with a dot.

**Comment by seanybaggins** [comment]
> Nit: please end full sentences in commit message bodies with a dot.

Fixed

**Review by bjornfor — APPROVED** [review]

**Review by NickCao — CHANGES_REQUESTED** [review]
`libwebp` and `libmng` all builds fine on mingw, only `jasper` lacks cross compilation support.

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingw32.qt6.qtsvg

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingwW64.qt6.qtsvg

File: pkgs/development/libraries/qt-6/modules/qtsvg.nix [hunk 1]
```diff
@@ -5,11 +5,20 @@
...
+, stdenv
 }:
 
 qtModule {
   pname = "qtsvg";
   propagatedBuildInputs = [ qtbase ];
-  buildInputs = [ libwebp jasper libmng zlib ];
+  buildInputs = [
+    libwebp
+  ] ++ lib.optionals (!stdenv.hostPlatform.isMinGW) [
```
Comment (NickCao) [inline pkgs/development/libraries/qt-6/modules/qtsvg.nix:17]:
```suggestion
  ] ++ lib.optionals (!stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
```

Comment (seanybaggins) [reply pkgs/development/libraries/qt-6/modules/qtsvg.nix:17]: This doesn't make sense to me. The buildPlatform can execute the hostPlatform? What does that mean?

Comment (NickCao) [reply pkgs/development/libraries/qt-6/modules/qtsvg.nix:17]: For example when you are cross compiling from x86_64-linux to i386-linux.

Comment (NickCao) [reply pkgs/development/libraries/qt-6/modules/qtsvg.nix:17]: But in most cases it just means cross compiling or not.

Comment (seanybaggins) [reply pkgs/development/libraries/qt-6/modules/qtsvg.nix:17]: Oh Okay that makes sense. I forget that stdenv here when I am trying to cross compile is pkgsCross.platform.stdenv

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingw32.qt6.qtsvg

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingwW64.qt6.qtsvg

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingw32.qt6.qtsvg

**Comment by seanybaggins** [comment]
@ofborg build pkgs.pkgsCross.mingwW64.qt6.qtsvg

**Review by NickCao — APPROVED** [review]

**Pull request merged**